content/articles/how-to-create-nlp-application-with-flair/index.md

```diff
@@ -0,0 +1,424 @@
+
+```python
+train,test,dev = np.split(df1,[int(.6*len(df1)),int(.8*len(df1))])
+```
+
+The above command allows us to split the dataset into three where 60% of the dataset will be used as a train set, 20% will be test set, and the remaining 20% will be dev set.
+
+#### Checking the train set, test set, and dev set
+After splitting we need to check the size of the three datasets.
+
+```python
+print(df1.shape)
+print(train.shape)
+print(test.shape)
+print(dev.shape)
+```
+
+This allows us to see the size of our train, test, and dev sets.
+The output is as shown.
+
+```bash
+(4671, 2)
+(2802, 2)
+(934, 2)
+(935, 2)
+```
+
+After splitting the data into three you need to store the data into a folder.
+
+#### Creating a folder
+Let's create a folder where we can store our three sets of data. Since we are using [Google Colab](https://research.google.com/), the following command will create a folder named `data_fst` for us.
+
+```python
+!mkdir -p data_fst
+```
+
+We then need to save the three sets of data into the created folder. The data sets are saved in a CSV format.
+
+```python
+train.to_csv("data_fst/train.csv")
+test.to_csv("data_fst/test.csv")
+dev.to_csv("data_fst/dev.csv")
+```
+
+### Building corpus
+A corpus is a collection of a large and structured set of machine-readable texts that represents the dataset that is used in building our model.
+It consists of a list of sentences that corresponds to the training, testing, and validation sets of our data that is used during model training.
+
+In this section, we shall use the `CSVClassificationCorpus` in creating our corpus. `CSVClassificationCorpus` allows us to build corpus using the CSV file format that we created earlier in this tutorial.
+NOTE > Since our dataset is prepared in form of a CSV we have to use `CSVClassificationCorpus` when building our corpus.
+
+We start by importing `Corpus` and `CSVClassificationCorpus`.
+
+```python
+import CSVClassificationCorpus from flair.datasets
+import Corpus from flair.data
+```
+
+After we have successfully imported `Corpus` and `CSVClassificationCorpus`, we have to create column mapping to show which column is for labels and text. As shown earlier in this tutorial our CSV dataset has two columns: 'labels and 'text'.
+We shall map our first column as a 'label_topic' and our second column as 'text' using the following command.
+
+```python
+column_name_map = {2:"label_topic",1:"text"}
+```
+
+Mapping the column name will enhance efficiency when building our model since the model will easily know which column to use as a label and which as a feature.
+We also need to give the location of our dataset so that the model can know where our data is located during training.
+
+```python
+data_folder = 'data_fst/'
+```
+
+After doing all this we can now create our corpus using the `CSVClassificationCorpus` since we are using a CSV file format hence we specify our `delimiter` as a `,`.
+
```

Comment:
Consider splitting this sentence into two.

After doing all this, we can now create our corpus using the `CSVClassificationCorpus`. Since we are using a CSV file format, we specify our `delimiter` as a `,`.